### ISSUE TITLE
Shader pack meta

### ISSUE THREAD

**Issue by SmajloSlovakian** [IrisShaders/Iris#996]
There are meta informations about resourcepacks, why not make some in shaderpacks too?
The way i think of it is that inside the root of shaderpack structure, there will be pack.mcmeta and pack.png. Png is self explanatory and the meta could obtain a description, officially supported versions*, hardware requirements*, features list, maybe an accent color which shows on the button and a link to official download page.
This should not affect optifine as you can add any file into the root directory.
if the shaderpack does not have any meta information, it can be considered as "may contain bugs" pack.

* this can lead to a warning that you don't meet certain requirements

**Comment by ColdIce1605** [comment]
Yes this would be good in the future, tho right now we're (they're) going for compatibility.

**Comment by SmajloSlovakian** [comment]
> Yes this would be good in the future, tho right now we're (they're) going for compatibility.

yeah i know that, this request is in fact for the future...

**Comment by SmajloSlovakian** [comment]
the Shader pack selection menu could then be tile-based:
![image](https://user-images.githubusercontent.com/16209307/143271163-ce61b6c5-da0e-4f2c-bde6-58e60b5d5251.png)
optifine shader packs may look a bit silly though

**Comment by coderbot16** [comment]
This could also have an option to allow a separate "display name" as well supporting formatting and color codes. Some shader pack authors have resorted to including Minecraft color codes in their file names to do this.
